pages/__admin/messages/index.tsx

```diff
@@ -0,0 +1,81 @@
+import AdminMessageItem from "feature/admin-messages-list/AdminMessageItem"
+import ChannelMessageFunction from "databases/functions/ChannelMessages"
+import getConfig from "next/config"
+import init from 'nextjs-basic-auth'
+import useToastHook from "@hooks/toast/useToastHook"
+import { AdminChannelMessage } from "types/Feature/AdminMessagesList/AdminMessageItem"
+import { GetServerSidePropsContext } from "next"
+import { getAllAdminMessages } from "@services/message/adminMessageService"
+import { useRouter } from "next/router"
+import { useState } from "react"
+
+export const getServerSideProps = async (context: GetServerSidePropsContext) => {
+    const {serverRuntimeConfig} = getConfig()
+    const {query} = context
+
+    const users = [
+        {user: serverRuntimeConfig.BETTER_BACKDOOR_ADMIN_USER, password: serverRuntimeConfig.BETTER_BACKDOOR_PASSWORD},
+    ]
+
+    const basicAuthCheck = init({
+        users
+    })
+
+    await basicAuthCheck(context.req, context.res)
+
+    const offset = query?.offset ? parseInt(query?.offset as string) : 0
+    const channelMessages = await ChannelMessageFunction.adminGetChannelMessage(10, offset)
+    
+    return {
+        props: {
+            messages: channelMessages
+        }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `getServerSideProps` function is using basic authentication to protect the admin page. However, storing credentials in environment variables and checking them directly in the code can be a security risk. It would be better to use a more secure method of authentication, such as OAuth or JWT.



<!-- This is an auto-generated comment by OSS CodeRabbit -->